src/Data/Rational/Unnormalised/Base.agda

```diff
@@ -134,15 +134,26 @@ p - q = p + (- q)
 
 -- reciprocal: requires a proof that the numerator is not zero
 
-1/_ : (p : ℚᵘ) → .{n≢0 : ∣ ↥ p ∣ ≢0} → ℚᵘ
+1/_ : (p : ℚᵘ) → .{n≢0 : ∣ ↥ p ∣ᶻ ≢0} → ℚᵘ
 1/ mkℚᵘ +[1+ n ] d = mkℚᵘ +[1+ d ] n
 1/ mkℚᵘ -[1+ n ] d = mkℚᵘ -[1+ d ] n
 
 -- division: requires a proof that the denominator is not zero
 
-_÷_ : (p q : ℚᵘ) → .{n≢0 : ∣ ↥ q ∣ ≢0} → ℚᵘ
+_÷_ : (p q : ℚᵘ) → .{n≢0 : ∣ ↥ q ∣ᶻ ≢0} → ℚᵘ
 (p ÷ q) {n≢0} = p * (1/_ q {n≢0})
 
+-- max
+_⊔_ : (p q : ℚᵘ) → ℚᵘ
+p ⊔ q = ((↥ p ℤ.* ↧ q) ℤ.⊔ (↥ q ℤ.* ↧ p)) / (↧ₙ p ℕ.* ↧ₙ q)
```

Comment:
Looking at the moderately scary equational proofs below, I'm wondering if this is the best definition. What are the advantages of this over something like the following?
```
p ⊔ q with does (↥ p ℤ.* ↧ q ℤ.≤? ↥ q ℤ.* ↧ p)
... | true  = p
... | false = q
```
I'd imagine the above is significantly easier to work with?

Reply:
Not really I think. It makes other properties much harder as instead of leveraging integer properties you need to split into multiple cases and prove that some of them does not apply. For example `⊓-comm` would have 4 cases and `⊓-assoc` would have 8. 

On the flip side the above property is proven by combination of:
```agda

p≤q⇒p⊓q≃p : p ≤ q → p ⊓ q ≃ p                                                                                                                                                       
q≤p⇒p⊓q≃q : q ≤ p → p ⊓ q ≃ q
≤-total : Total _≤
```

I will think if the reasoning can be replaced by those properties.

Reply:
Hmm I'm wondering how we can somehow get away with reusing the `Min` and `Max` code here:

https://github.com/agda/agda-stdlib/blob/master/src/Algebra/Construct/NaturalChoice/Min.agda
https://github.com/agda/agda-stdlib/blob/master/src/Algebra/Construct/NaturalChoice/Max.agda

It's tricky though as even accessing the raw definition requires providing a total order, and `Total` proofs aren't designed to be implemented efficiently...

Reply:
Just a heads up that I have a way of getting many of the proofs about min and max for free. Going to take a little while to get it working, but just a heads up that I haven't forgotten this!

Reply:
Just coming back to this discussion now that we get most of the properties for free, I really do think the that the definition I propose above is better. Firstly it avoids unnecessarily blowing up the numerator and denominator, and secondly it means that you can work with propositional equality rather than `_≃_`.